src/types.ts

```diff
@@ -9,6 +9,7 @@ export interface SessionCreatedParameters {
 	sessionId: string;
 	rootNode: NodeInfo;
 	errorMessage: string;
+	errorNumber: number;
```

Comment:
Let's take the opportunity to add doc comments and fix up the typings here too.

(and then also update ObjectExplorerSession in azdata.d.ts)

(Note that technically rootNode is also optional, but I'd rather leave that as required for the time being since it should always be included if the connection was a success which isn't necessarily true for the error message/code. Those are optional even if it fails)

```ts
export interface SessionCreatedParameters {
	/**
	 * Whether the session was created successfully.
	 */
	success: boolean;
	/**
	 * The ID of the session.
	 */
	sessionId: string;
	/**
	 * The root node for the session, if it was created successfully.
	 */
	rootNode: NodeInfo;
	/**
	 * Error message for the failure, if the session was not created successfully.
	 */
	errorMessage?: string | undefined;
	/**
	 * Error number for the failure, if the session was not created successfully.
	 */
	errorNumber?: number | undefined;
}
```